main/webapp/modules/core/scripts/index.js

```diff
@@ -68,28 +68,30 @@ Refine.selectActionArea = function(id) {
 };
 
 $(function() {
-  var isThereNewRelease = function() {
-    // TODO: This eeds to be modified to do checking based solely on version, not revision
-    var thisRevision = OpenRefineVersion.revision;
-    var thisVersion = OpenRefineVersion.version;
-
-    var revision_pattern = /r([0-9]+)/;
-
-    if (!revision_pattern.test(thisRevision)) { // probably "trunk"
+  var isThereNewRelease = function(thisVer,latestVer) {
+    // Assumes Semantic Version numbering format of form X.Y.Z where X, Y, and Z are non-negative integer
+    // Ignores any trailing letters e.g. 2.7.1-rc.1 will be interpreted as equivalent to 2.7.1
+    if(thisVer == latestVer) {
       return false;
     }
-
-    var latestRevision = OpenRefineReleases.releases[0].revision;
-    var latestVersion = OpenRefineReleases.releases[0].version;
-
-    var thisRev = parseInt(revision_pattern.exec(thisRevision)[1],10);
-    var latestRev = parseInt(revision_pattern.exec(OpenRefineReleases.releases[0].revision)[1],10);
-
-    // Parser version into main version and suffix which follows dash
-    // Parse version from dot separated string into array of integers
-    
-    // compare left to right, including suffix
-    return latestRev > thisRev;
+    var thisVerParts = thisVer.split(".");
+    var latestVerParts = latestVer.split(".");
+    while(thisVerParts.length < 3)
+      thisVerParts.push(0);
+    while(latestVerParts.length < 3)
+      latestVersParts.push(0);
+
+    for(var i=0; i<3; i++) {
+      console.log(parseInt(latestVerParts[i],10));
+      if(parseInt(thisVerParts[i],10) == parseInt(latestVerParts[i],10)) {
```

Comment:
Why not store `parseInt(thisVerParts[i],10)` in a variable and do the comparisons on that variable? Same for `parseInt(latestVerParts[i],10)`, it would look a bit cleaner to parse the string only once.

Reply:
OK